ts/WoltLabSuite/WebComponent/fa-icon.ts

```diff
@@ -0,0 +1,138 @@
+(() => {
+  let isFA6Free: boolean;
+  function isFontAwesome6Free(): boolean {
+    if (isFA6Free === undefined) {
+      isFA6Free = true;
+
+      const iconFont = window.getComputedStyle(document.documentElement).getPropertyValue("--fa-font-family");
+      if (iconFont === "Font Awesome 6 Pro") {
+        isFA6Free = false;
+      }
+    }
+
+    return isFA6Free;
+  }
+
+  const HeightMap = new Map<number, number>([
+    [16, 14],
+    [24, 18],
+    [32, 28],
+    [48, 42],
+    [64, 56],
+    [96, 84],
+    [128, 112],
+    [144, 130],
+  ]);
+
+  class FaIcon extends HTMLElement {
+    connectedCallback() {
+      this.validate();
+      this.setIcon(this.name, this.solid);
+    }
+
+    private validate(): void {
+      if (this.size === 0) {
+        throw new TypeError("Must provide an icon size.");
+      } else if (!HeightMap.has(this.size)) {
+        throw new TypeError("Must provide a valid icon size.");
+      }
+
+      if (this.name === "") {
+        throw new TypeError("Must provide the name of the icon.");
+      } else if (!this.isValidIconName(this.name)) {
+        throw new TypeError(`The icon '${this.name}' is unknown or unsupported.`);
+      }
+    }
+
+    setIcon(name: string, isSolid: boolean): void {
+      if (!this.isValidIconName(name)) {
+        throw new TypeError(`The icon '${name}' is unknown or unsupported.`);
+      }
+
+      if (!this.isValidIconStyle(name, isSolid)) {
+        throw new Error(`The icon '${name}' only supports the 'solid' style.`);
+      }
+
+      this.solid = isSolid;
+      this.name = name;
+
+      this.updateIcon();
+    }
+
+    private isValidIconName(name: string | null): boolean {
+      return name !== null && window.getFontAwesome6IconMetadata(name) !== undefined;
+    }
+
+    private isValidIconStyle(name: string, isSolid: boolean): boolean {
+      if (!isSolid && isFontAwesome6Free()) {
+        const [, styles] = window.getFontAwesome6IconMetadata(name)!;
+        if (!styles.includes("regular")) {
+          return false;
+        }
+      }
```

Comment:
This part would benefit from a short (!) explanatory comment:

> We must reject non-solid icons that are not part of FA 6 Free.

or something like this.